staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -984,6 +1019,10 @@ func (p *processorListener) run() {
 			default:
 				utilruntime.HandleError(fmt.Errorf("unrecognized notification: %T", next))
 			}
+			p.metrics.processDuration.Observe(time.Since(startTime).Seconds())
+			//TODO: This requires implementing Len() and Capacity() for ring growing
+			// p.metrics.numberOfPendingNotifications.Set(float64(p.pendingNotifications.Len()))
+			// p.metrics.sizeOfRingGrowing.Set(float64(p.pendingNotifications.Capacity()))
```

Comment:
We need to wait for the `Len()` and `Capacity()` methods in the ring growing package to be merged.
PR: https://github.com/kubernetes/utils/pull/321

Reply:
is this single-threaded? (is calling Len and Capacity independently and not under lock safe here, given the `pendingNotifications` is not thread-safe?)

Reply:
done